src/ios.ts

```diff
@@ -497,6 +579,55 @@ Components:" > /etc/apt/sources.list.d/appstraction.sources`,
         )
             throw new Error('Failed to set proxy.');
     },
+    addCalendarEvent: async (eventData) => {
+        if (!options.capabilities.includes('frida'))
+            throw new Error('Frida is required to add events to the calendar.');
+
+        // The ObjC formatter does not understand milliseconds
+        const simplifiedISO = (date: Date) => date.toISOString().replace(/\.[0-9]{0,3}Z$/, 'Z');
+
+        await frida.getUsbDevice().then(async (f) => {
+            // frida has problems spawning a process to attach to on iOS
+            await f.attach('SpringBoard').then(async (session) => {
+                const startCalendar = await session.createScript(fridaScripts.startApp('com.apple.mobilecal'));
+                await startCalendar.load();
+                await session.detach();
+            });
+            // wait until the Calendar has started
+            await pause(3000);
```

Comment:
This seems flaky. Can't we use `getForegroundAppId()` and `retryCondition()` instead?

Reply:
You can try, but it might just be as flaky, because the app needs to start and then it also needs to display the GUI for the intent, which might take longer than expected.